src/MenuCache.php

```diff
@@ -82,6 +82,8 @@ public function get_menu( $menu, $args ) {
 		if ( $this->should_cache_menu( $args ) && isset( $args->{$this->key_argument} ) ) {
 			$cached_menu = get_transient( $args->{$this->key_argument} );
 			if ( is_string( $cached_menu ) ) {
+				// We have found what we need, so we don't want any code to run after us (since any changes made here would be impossible to restore)
+				remove_all_filters( 'pre_wp_nav_menu' );
```

Comment:
What would such an incorrect way be? 

I agree this feels like a drastic measure, but I am not creative enough to find a potential problem with killing all other callbacks once we have a cache hit. 

The intended use of this hook is to return HTML to render instead of creating the menu from scratch. 

So people will either return full HTML (instead of adding to it) or they're misusing the hook. 

The only thing we could run in trouble with with is another caching plugin (in which case users might not even care as long as there is a cache) 


What would be a use case for allowing code to run after we're done?